community/cypher/cypher/src/test/scala/org/neo4j/cypher/EagerizationAcceptanceTest.scala

```diff
@@ -2323,6 +2323,39 @@ class EagerizationAcceptanceTest
     assertNumberOfEagerness(query, 2, optimalEagerCount = 1)
   }
 
+  test("should always be eager after deleted relationships if there are any subsequent expands that might load them") {
+    val device = createLabeledNode("Device")
+    val cookies = (0 until 2).foldLeft(Map.empty[String, Node]) { (nodes, index) =>
+      val name = s"c$index"
+      val cookie = createLabeledNode(Map("name" -> name), "Cookie")
+      relate(device, cookie)
+      relate(cookie, createNode())
+      nodes + (name -> cookie)
+    }
+
+    val query =
+      """
+        |MATCH (c:Cookie {name: {cookie}})<-[r2]-(d:Device)
+        |WITH c, d
+        |MATCH (c)-[r]-()
+        |DELETE c, r
+        |WITH d
+        |MATCH (d)-->(c2:Cookie)
+        |RETURN d, c2""".stripMargin
+
+    assertNumberOfEagerness(query, 2, optimalEagerCount = 1)
```

Comment:
I'm not certain 1 is possible for this query. There needs to be an eager between `MATCH (c)-[r]-()` and `DELETE c, r`, due to the the match being undirected. And there needs to be an eager between `DELETE c, r` and `MATCH (d)-->(c2:Cookie)` due to the match being unstable.